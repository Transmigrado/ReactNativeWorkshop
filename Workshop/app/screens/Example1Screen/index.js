
import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'


const Example1Screen = ({ navigation, route }) => {
    
    const onPress =  () => {
        navigation.navigate('Example1')
    }

    return (
        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
            <Text>Primera pantalla</Text>
                <TouchableOpacity id="mainButton" onPress={onPress}>
                    <Text>Ir al Detalle</Text>
                </TouchableOpacity>
        </View>
    )
}

export default Example1Screen
