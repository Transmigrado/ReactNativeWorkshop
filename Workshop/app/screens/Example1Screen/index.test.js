

import React from 'react'
import renderer from 'react-test-renderer'
import Example1Screen from './index'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({adapter: new Adapter()})

describe('testeaos snapshot', () => {
    it('renders correctly for styleButton is STYLE_1', () => {
        const tree = renderer
                        .create(<Example1Screen />)
                        .toJSON()
    
        expect(tree).toMatchSnapshot()
    })
    
    
})


it('simulamos presionar el boton y navegamos a Example2', () => {

    const navigation = {
        navigate: jest.fn()
    }

    const wrapper  = shallow(<Example1Screen navigation={navigation} />)
    const button = wrapper.find('#mainButton')
    button.simulate('press')

    expect(navigation.navigate).toHaveBeenCalledWith('Example1')

})