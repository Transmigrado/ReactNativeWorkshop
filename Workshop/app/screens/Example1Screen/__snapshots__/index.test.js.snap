// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testeaos snapshot renders correctly for styleButton is STYLE_1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Primera pantalla
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Ir al Detalle
    </Text>
  </View>
</View>
`;
